sopel/formatting.py

```diff
@@ -124,7 +125,34 @@
 PLAIN_REGEX = re.compile(PLAIN_PATTERN)
 
 
-class colors(str, Enum):
+class MatchText(enum.Flag):
```

Comment:
Can it not live in `sopel.plugin`? Doesn't look like `sopel.plugins` or `sopel.plugins.rules` gets imported by that, so it shouldn't create any dependency loop problem. Living there makes sense if plugin authors are meant to use the Enum as an argument to a decorator from that same module, anyway.

Reply:
I *was* reluctant to put it in `sopel.plugin` (emphasis en "was"), because it means some kind of dependencies between `sopel.plugins.*` to it... but it's actually an acceptable dependency. Same for `sopel.loader`, I don't mind anymore. So I've moved it to `sopel.plugin` (not pushed yet at the time of writing this comment).